include/seastar/http/client.hh

```diff
@@ -220,7 +220,14 @@ public:
      * may re-use the sockets on its own
      *
      * \param f -- the factory pointer
+     * \param max_connections -- maximum number of connection a client is allowed to maintain
```

Comment:
I don't understand why we wanted such a feature... What happens if the user starts another request? It waits until one of the older request finishes? Why?

Anyway, you're just documenting a feature we already have, so ok.

I can understand a maximum number of connections **cached** in the pool, that's useful. But why do we want to artificially limit the total number of connections?

Although I guess you're just documenting this now,

Reply:
> What happens if the user starts another request? It waits until one of the older request finishes? Why?

If `max_connections` is already serving other requests (and pool is empty) then yes. Because this knob limits the number of _sockets_ opened simultaneously. Limiting only the number of connections in pool is also useful, but I think concurrency control is also required for active connections (even though they are supposedly short-living)

Reply:
And yes, it's the pre-existing feature from #1652  (you approved it back then :wink:  )